### PR TITLE
chore: Using mergeWith to manage labels

### PR DESCRIPTION
**Pull request by nsainaney**
Depends on https://github.com/c6o/kubeclient/pull/15

### CODE REVIEW

File: packages/appengine/src/index.ts [hunk 1]
```diff
@@ -16,10 +16,10 @@ export interface Provisioner extends ProvisionerBase { }
...
 export class Provisioner extends mix(ProvisionerBase).with(createApplyMixin, createInquireMixin, templateHelperMixin) {
 
-    _manifestHelper
-    get manifestHelper(): AppEngineAppObject {
-        if (this._manifestHelper)
-            return this._manifestHelper
-        return this._manifestHelper = new AppEngineAppObject(this.manager.document)
+    // Override the documentHelper in ProvisionerBase
+    get documentHelper(): AppEngineAppObject {
+        if (this._documentHelper || !this.manager?.document)
```
Comment (Canuckaholic) [inline packages/appengine/src/index.ts:21]: extremely minor thing but I don't think you need the `?` behind a _not_ condition:
`!this.manager.document`

File: packages/appengine/src/mixins/createApply.ts [hunk 2]
```diff
@@ -86,27 +86,27 @@ export const createApplyMixin = (base: baseProvisionerType) => class extends bas
...
                 skipped = true
                 return
             }
 
             const configs = {}
-            for(const key of Object.keys(this.manifestHelper.configs)) {
+            for(const key of Object.keys(this.documentHelper.configs)) {
                 // Must be string, even if originally a boolean or number.
-                configs[key] = String(this.manifestHelper.configs[key])
+                configs[key] = String(this.documentHelper.configs[key])
```
Comment (Canuckaholic) [inline packages/appengine/src/mixins/createApply.ts:96]: I think either of these would work as well:
```
for (const [key, value] of Object.entries(this.documentHelper.configs)) {
  configs[key] = String(value)
}
```
or
```
for (const key in this.documentHelper.configs) {
  configs[key] = String(this.documentHelper.configs[key])
}
```

File: packages/common/src/provisioner.ts [hunk 3]
```diff
@@ -29,10 +30,17 @@ export class ProvisionerBase extends mix(provisionerBasePrivate).with(namespaceM
...
     [key: string]: any
 
-    help (command: string, options: optionFunctionType, messages: string[]) {}
+    help (command: string, options: optionFunctionType, messages: string[]) { }
 
     get edition(): string { return this.manager?.document?.metadata?.labels['system.codezero.io/edition'] }
 
+    _documentHelper
+    get documentHelper(): AppObject {
+        if (this._documentHelper || !this.manager?.document)
```
Comment (Canuckaholic) [inline packages/common/src/provisioner.ts:39]:
```suggestion
        if (this._documentHelper || !this.manager.document)
```

**Review by Canuckaholic — APPROVED** [review]

File: packages/appengine/src/index.ts (hunk 1 above)
Comment (nsainaney) [reply packages/appengine/src/index.ts:21]: If manager is null/undefined, you'll get an exception when trying to access document so you do need the ?

File: packages/common/src/provisioner.ts (hunk 3 above)
Comment (nsainaney) [reply packages/common/src/provisioner.ts:39]: this.manager may be undefined

**Review by nothingmn — APPROVED** [review]

**Pull request merged**